README.md
# Données publiques
Ces données sont publiques, elle sont utilisées dans le cadres du cours **Laboratoire d'actuariat (ACT3035)**.


CASdatasets/NEWS
=== CASdatasets: Datasets for the book 'Computational Actuarial Science with R' ===


Version 0.9-8
=============

- initial release. 


Dataset list with entry date
=============

- lossalae 30/08/2012
- danishuni 12/10/2013
- danishmulti 12/10/2013
- auscathist 19/12/2012
- itamtplcost 03/01/2013
- tplclaimnumber 30/04/2013
- canlifins 20/06/2013
- freclaimset 20/06/2013
- freaggnumber 20/06/2013
- freMTPLfreq 21/06/2013
- freMTPLsev 21/06/2013
- freTD8890 24/06/2013
- freTV8890 24/06/2013
- freTF0002 24/06/2013
- freTH0002 24/06/2013
- freTPRV93 24/06/2013
- freAS0002 24/06/2013
- freTGF05 25/06/2013
- freTGH05 25/06/2013
- freTPG93full 25/06/2013
- hurricanehist 25/06/2013
- frebiloss 25/06/2013
- ICB1 13/09/2013
- ICB2 13/09/2013
- frecomfire 15/09/2013
- Davis 17/09/2013
- Norberg 17/09/2013
- beaonre 18/09/2013
- lossalaefull 19/09/2013
- besecura 19/09/2013
- SOAGMI 19/09/2013
- norfire 19/09/2013
- autoBI 19/09/2013
- sgautonb 19/09/2013
- swautoins 19/09/2013
- ushuannualloss 28/09/2013
- ushuinflation 28/09/2013
- ushupopulation 28/09/2013
- ushustormloss 28/09/2013
- credit 02/01/2014
- ukaggclaim 09/01/2014
- frefictivetable 10/01/2014
- freprojqxINSEE 10/01/2014
- spacedata 16/01/2014
- usworkcomp 16/01/2014
- usexpense 17/01/2014
- usquakeLR 20/01/2014
- ausNLHYFinPerf 13/02/2014
- ausNLHYFinPos 13/02/2014
- ausNLHYLloydAsset 13/02/2014
- ausNLHYLloydGPI 13/02/2014
- ausNLHYLloydUWAcc 13/02/2014
- ausNLHYLloydUWRes 13/02/2014
- ausNLHYOffProf 13/02/2014
- ausNLHYOpIncExp 13/02/2014
- ausNLHYRecAASB 13/02/2014
- ausNLHYReserve 13/02/2014
- ausNLHYCapAdeq 13/02/2014
- ausNLHYCapAdeqByComp 13/02/2014
- ausNLHYClaimByState 13/02/2014
- ausNLHYFinPerfByComp 13/02/2014
- ausNLHYFinPerfPublic 13/02/2014
- ausNLHYFinPosByComp 13/02/2014
- ausNLHYFinPosPublic 13/02/2014
- ausNLHYLiability 13/02/2014
- ausNLHYOpIncExpPublic 13/02/2014
- ausNLHYPremByState 13/02/2014
- ausNLHYPremClaim 13/02/2014
- ausNLHYPremClaimPublic 13/02/2014
- ausNLHYPrivInsur 13/02/2014
- ausNLHYPubInsur 13/02/2014
- norauto 05/03/2014
- ushustormloss4980 07/03/2014
- linearmodelfactor 28/03/2014
- freptfpermdis 24/04/2014
- freptftempdis 24/04/2014
- frepermdis10 24/04/2014 (removed)
- fretempdis10 24/04/2014 (removed)
- fredisprob10 24/04/2014 (removed)
- eqlist 30/04/2014
- sgautoprop9701.rda 08/09/2014
- ukautocoll.rda 08/09/2014
- usautoBI.rda 08/09/2014
- usmassBI.rda 08/09/2014
- usprivautoclaim.rda 08/09/2014
- usreGL8190.rda 08/09/2014
- usreGL8700.rda 08/09/2014
- ustermlife.rda 08/09/2014
- uswarrantaggnum.rda 08/09/2014
- freMPL1.rda 09/09/2014
- freMPL2.rda 09/09/2014
- freMPL3.rda 09/09/2014
- freMPL4.rda 09/09/2014
- freMPL5.rda 09/09/2014
- freMPL6.rda 09/09/2014
- freMPL7.rda 09/09/2014
- freMPL8.rda 09/09/2014
- freMPL9.rda 09/09/2014
- freMPL10.rda 09/09/2014
- fremotor2claim.rda 11/09/2014
- usmedclaim.rda 11/09/2014
- sgautoBI9301.rda 11/09/2014
- usautotri9504.rda 11/09/2014
- fretri4auto9403.rda 25/09/2014
- fretri1auto9605.rda 25/09/2014
- fretri2auto9605.rda 25/09/2014
- fretri3auto9605.rda 25/09/2014
- austri1autoBI7895.rda 01/10/2014
- usmassBI2.rda 12/10/2014
- freMTPL2sev.rda 14/10/2014
- freMTPL2freq.rda 14/10/2014
- nzcathist.rda 05/12/2014
- swbusscase.rda 11/12/2014
- swmotorcycle.rda 11/12/2014
- ausprivautolong.rda 12/12/2014
- austri2auto.rda 12/12/2014
- ausautoBI8999.rda 12/12/2014
- ausMTPL8486.rda 12/12/2014
- ausprivauto0405.rda 12/12/2014
- ausNSWdriver04.rda 12/12/2014
- ausNSWdeath02.rda 12/12/2014
- forexUSUK.rda 13/01/2015
- freP2Pdis10.rda 16/06/2015
- freT2Tdis10.rda 16/06/2015
- freT2Pdis10.rda 16/06/2015
- freT2Ddis10.rda 16/06/2015
- freP2Ddis10.rda 16/06/2015
- freT2Pdisprob10.rda 16/06/2015
- freT2Tdisprob10.rda 16/06/2015
- freP2Pdisprob10.rda 16/06/2015
- freP2Ddisprob10.rda 16/06/2015
- freT2Ddisprob10.rda 16/06/2015
- nortritpl8800.rda 28/08/2015
- asiacomrisk.rda 14/12/2015
- brvehins2a,b,c,d.rda 27/03/2016
- braggclaim.rda 27/03/2016
- braggprem.rda 27/03/2016
- brvehins1a,b,c,d,e.rda 27/03/2016
- brgeomunic.rda 27/03/2016
- brgeomunicins.rda 27/03/2016
- brautocoll.rda 27/03/2016
- fremotor1prem.rda 04/04/2016
- fremotor1freq.rda 04/04/2016
- fremotor1sev.rda 04/04/2016
- frePF6064.rda 28/05/2016
- frePM6064.rda 28/05/2016
- freTD7377.rda 28/05/2016
- freTV7377.rda 28/05/2016
- fremotor2sev9907.rda 28/05/2016
- fremotor3sev9907.rda 28/05/2016
- fremotor4sev9907.rda 28/05/2016
- fremotor4freq9907b.rda 28/05/2016
- fremotor2freq9907b.rda 28/05/2016
- fremotor3freq9907b.rda 28/05/2016
- fremotor4freq9907u.rda 28/05/2016
- fremotor2freq9907u.rda 28/05/2016
- fremotor3freq9907u.rda 28/05/2016


codeBrut/cours_9_2.r

options(repr.matrix.max.cols=8, repr.matrix.max.rows=8) #seulement pour afficher 8 lignes et 8 colonnes

# test_csv <-read.csv("https://raw.githubusercontent.com/nmeraihi/data/master/exemple_1.csv")
test_csv <-read.csv("exemple_1.csv")
test_csv

test_csv <-read.csv("https://raw.githubusercontent.com/nmeraihi/data/master/exemple_1.csv")
test_csv

attributes(test_csv)

test_csv <-read.csv("exemple_1.csv", header = F)
test_csv

exemple <-read.csv("exemple_1.csv", header = T)
exemple

class(exemple$Segment)

exemple <-read.csv("exemple_1.csv", header = T, stringsAsFactors=F)
exemple

class(exemple$Segment)

exemple <-read.csv("exemple_1.csv", header = T, stringsAsFactors=F)[,2:3]
exemple

exemple <-read.csv("exemple_1.csv", header = T, stringsAsFactors=F)[,c(1,3)]
exemple

read.table("exemple_1.txt", header=T)

read.table("exemple_1.txt", header=T, sep = ",")

install.packages("RCurl")

library(RCurl)

x <- getURL("https://raw.githubusercontent.com/aronlindberg/latent_growth_classes/master/LGC_data.csv")
y <- read.csv(text = x)

head(y)

read.table("https://raw.githubusercontent.com/nmeraihi/data/master/exemple_2_2.txt", header=T, sep = ",")

df<-read.table("https://raw.githubusercontent.com/nmeraihi/data/master/exemple_2_2.txt", header=T, sep = ",")
is.na(df)

is.na(df[1,1])

is.na(df[2,1])

mean(df$km)

mean(na.omit(df$km))

na.omit(df$km)

mean(df$km,na.rm = T)

na.omit(df)

 df[2,1]<-4.84
 df[4,4]<-125

df

df<-read.table("https://raw.githubusercontent.com/nmeraihi/data/master/exemple_2_2.txt", header=T, sep = ",")
df

df[is.na(df)] <- 0

df


codeBrut/Exercices_10_sol.r

options(repr.matrix.max.cols=8, repr.matrix.max.rows=5)

path<-"https://raw.githubusercontent.com/nmeraihi/data/master/"

df<-read.csv(paste(path,"qc_hommes_2.csv",sep = ""), sep=",")

head(df)

tail(df)

df$age<-gsub("ans", "", df$age)
df$age<-gsub("an", "", df$age)

a<-df[-nrow(df), 2]-df[-1, 2]
a<-c(a, a[length(a)])
df$dx<-a

df$qx<-round(df$dx/df$lx,5)

head(df)

age<-22
t<-3
p<-df[age+1+t,2]/df[age+1,2]
p

library(formattable)

percent(p)

Id=c(1,2,3,4)
Age=c(14,12,15,10)
Sex=c('F','M','M','F')
Code=c('a','b','c','d')
df1=data.frame(Id,Age)
df2=data.frame(Id,Sex,Code)

df1

df2

M=merge(df1,df2,by='Id')
M

path<-"https://raw.githubusercontent.com/nmeraihi/data/master/"

df_app <-read.csv(paste(path,"AAPL_month.csv",sep = ""), header = T)

df_ba <-read.csv(paste(path,"BA_month.csv",sep = ""), header = T)

a<-cbind(df_app$Adj.Close,df_ba$Adj.Close)

colnames(a)<-c("Apple", "Boeing")

rownames(a)<-seq(as.Date("2016/11/1"), by = "month", length.out = 12)

cor(a)

library(dplyr, warn.conflicts = F)

values <- read.csv(paste(path,"HackerRank-Developer-Survey-2018-Values.csv",sep = ""), header = T, stringsAsFactors = F)

head(values)

values_2<-values %>% 
    group_by(q3Gender) %>%
        filter(q3Gender %in% c('Male','Female'))%>% 
            count()

values_2$n<-(values_2$n/ sum(values_2$n)) * 100
values_2

values$is_student <- ifelse(values$q8Student == '','Yes','No')

values %>% group_by(q3Gender, is_student) %>% 
    filter(q3Gender %in% c('Male','Female')) %>% 
    count() %>% 
    ungroup() %>% 
    group_by(is_student) %>% 
    mutate(n = (n/ sum(n)) * 100)

values %>% group_by(CountryNumeric2) %>% count() %>% head()

values %>%
            filter(!is.na(q4Education))%>%
            group_by(q4Education)%>%
            summarise(Total = n())%>%
            arrange(desc(Total))  %>%
            mutate(q4Education = reorder(q4Education,Total)) %>%
            head(10)

values %>%
            filter(!is.na(q1AgeBeginCoding)) %>%
            group_by(q1AgeBeginCoding) %>%
            summarise(Total = n()) %>%
            arrange(desc(Total)) %>%
            mutate(q1AgeBeginCoding = reorder(q1AgeBeginCoding,Total)) %>%
            head(10)


codeBrut/Exercices_8_solutions.r

pi

round(pi,3)

round(pi, 5)

vec1<-1:12
vec1

vec1<-c(vec1, 16:18)
vec1

seq(0,5,.5)

vec3<-seq(2,50, 2)
vec3

1:8

A<- matrix(c(1:8), nrow = 2, ncol = 4)
A

B<- matrix(c(1:8), nrow = 2, ncol = 4, byrow = T)
B

matrix(vec3, nrow = 5)

mat<-diag(7)
mat

apply(diag(7), 2, rev)

mat <- mat[ nrow(mat):1, ]
mat

pop<-matrix(c(1718241,1735096,1746940,1767753,
              530474,533857,536013,538918,
              417325,421959,425481,429413,
              274180,276290,278050,281392), ncol=4, byrow = T)

pop

rownames(pop)<-c("Montréal", "Québec", "Laval", "Gatineau")
colnames(pop)<-2013:2016

pop

vec_HOMA<-c(20000, 20500, 23000, 23800)
HOMA<-matrix(vec_HOMA, ncol=4, )
rownames(HOMA)<-c("HOMA")
colnames(HOMA)<-2013:2016
HOMA

pop[1,]-vec_HOMA

pop[1,]<-pop[1,]-vec_HOMA
pop

rbind(pop,HOMA)


codeBrut/cours_9_1.r

A<- matrix(c(6,8,1,1,4,2), nrow = 2, ncol = 3)

A

t(A)

vec<-1:5

t(vec)

dim(vec)

dim(t(vec))

A[1,3]

A

A[1,4]

A[,1]

B<-matrix(seq(1,9.5,.5), 3)

B

B[,c(2,4)]

B<-t(B)

B

B[c(3,5),]

diag(5)

A**2

B/2

C<-B*2

B+C

A<-matrix(c(30,32,31,27,36,72,60,78,67,71,55,57,56,55,49),ncol=3)
A

A<-round((A-32)/1.8,0)
A

colnames(A)<-c("Fairbanks","San Francisco","Chicago")

rownames(A)<-paste("3/",12:16,sep='')

paste("3/",12:16,sep='___')

A

B<-matrix(c(88,85,83,81,78,62,61,54,60,65,90,92,91,89,90),ncol=3)
colnames(B)<-c("Los Angeles","Seattle","Honolulu")
rownames(B)<-paste("3/",12:16,sep='')

B<-round((B-32)/1.8,0)
B

cbind(A,B)

rbind(A,B)

mat_comb<-cbind(A,B)

c(mat_comb)

mat_comb

min(mat_comb)

max(mat_comb)

range(mat_comb)

sd(mat_comb)

rowMeans(mat_comb)

colMeans(mat_comb)

cor(mat_comb)

summary(mat_comb)

summary(t(mat_comb))

runif(1,0,10)

runif(10,0,1)

x<-runif(100,0,1)
x

x<-matrix(x, 10)
x

set.seed(2)

runif(2,0,1) 

runif(2,0,1) 

runif(2,0,1) 

set.seed(2)

runif(2,0,1) 

runif(2,0,1) 

runif(2,0,1) 

sample(1:10,5)

sample(x,5)

sample(x,5)

sample(x,length(x)+1)

sample(0:1, 100, replace = T)

sample(state.name, 5)

s<-sample(1:5, 1000, replace=T, prob=c(.2,.2,.2,.2,.2))
s

table(s)

s<-sample(1:5, 1000, replace=T, prob=c(.2,.5,.1,.1,.1))
table(s)

rnorm(1)

rnorm(1, 0, 100)

x<-rnorm(100, 0, 100)

mean(x)

sd(x)

x<-rnorm(10000, 0, 100)
mean(x)

sd(x)

x<-array(1:24, dim=c(3,4,2))

x

(x <- list(taille = c(1, 5, 2), utilisateur = "Mike", new = TRUE))

x[1]

x[[1]]

x$taille

x$utilisateur

villes <-c("Montréal", "Québec", "Laval")
Population <-c(1942044, 585485, 430077)
village <-c(F,T,T)

donnees_ville <-data.frame(villes, Population, village)

donnees_ville

attributes(donnees_ville)

 donnees_ville[,1]

 donnees_ville[,2]

 donnees_ville[2,1]

donnees_ville$Population

 donnees_ville <-data.frame(villes, Population, village, stringsAsFactors=F)

donnees_ville

 donnees_ville$villes   

str(donnees_ville)

df<-donnees_ville

summary(df)

require(MASS)

data(Cars93)

class(Cars93)

str(Cars93)
